docs/hello_nextflow/04_hello_science.md

```diff
@@ -0,0 +1,763 @@
+Let's try running the workflow now that the plumbing is set up to run on all three test samples.
+
+```bash
+nextflow run hello-gatk.nf -resume
+```
+
+Funny thing: this might work, OR it might fail with an error like this:
+
+```console title="Output"
+ N E X T F L O W   ~  version 24.02.0-edge
+
+ ┃ Launching `hello-gatk.nf` [loving_pasteur] DSL2 - revision: d2a8e63076
+
+executor >  local (4)
+[01/eea165] SAMTOOLS_INDEX (2)       | 3 of 3, cached: 1 ✔
+[a5/fa9fd0] GATK_HAPLOTYPECALLER (3) | 1 of 3, cached: 1
+ERROR ~ Error executing process > 'GATK_HAPLOTYPECALLER (2)'
+
+Caused by:
+  Process `GATK_HAPLOTYPECALLER (2)` terminated with an error exit status (2)
+
+Command executed:
+
+  gatk HaplotypeCaller         -R ref.fasta         -I reads_father.bam         -O reads_father.bam.vcf         -L intervals.bed         -ERC GVCF
+
+Command exit status:
+  2
+
+Command error:
+```
+
+And further down, buried in the GATK command error output, there will be a line like this:
+
+```console title="Output"
+A USER ERROR has occurred: Traversal by intervals was requested but some input files are not indexed.
+```
+
+Well, that's weird, considering we explicitly indexed the BAM files in the first step of the workflow. Could there be something wrong with the plumbing?
+
+#### 3.2.1. Check the work directories for the relevant calls
+
+Let's take a look inside the work directory listed in the console output.
+
+```console title="Directory contents"
+work/a5/fa9fd0994b6beede5fb9ea073596c2
+├── intervals.bed -> /workspace/gitpod/hello-nextflow/data/ref/intervals.bed
+├── reads_father.bam.bai -> /workspace/gitpod/hello-nextflow/work/01/eea16597bd6e810fb4cf89e60f8c2d/reads_father.bam.bai
+├── reads_son.bam -> /workspace/gitpod/hello-nextflow/data/bam/reads_son.bam
+├── reads_son.bam.vcf
+├── reads_son.bam.vcf.idx
+├── ref.dict -> /workspace/gitpod/hello-nextflow/data/ref/ref.dict
+├── ref.fasta -> /workspace/gitpod/hello-nextflow/data/ref/ref.fasta
+└── ref.fasta.fai -> /workspace/gitpod/hello-nextflow/data/ref/ref.fasta.fai
+```
+
+Pay particular attention to the names of the BAM file and the BAM index that are listed in this directory: `reads_son.bam` and `reads_father.bam.bai`.
+
+What the heck? Nextflow has staged an index file in this process call's work directory, but it's the wrong one. How could this have happened?
+
+#### 3.2.1. Use the [view() operator](https://www.nextflow.io/docs/latest/reference/operator.html#view) to inspect channel contents
+
+Add these two lines in the workflow body before the `GATK_HAPLOTYPER` process call:
+
+```groovy title="hello-gatk.nf" linenums="84"
+    // temporary diagnostics
+    reads_ch.view()
+    SAMTOOLS_INDEX.out.view()
+```
+
+Then run the workflow command again.
+
+```bash
+nextflow run hello-gatk.nf
+```
+
+You may need to run it several times for it to fail again; this error will not reproduce consistently because it is dependent on some variability in the execution times of the individual process calls.
```

Comment:
semi-colon to two sentences
```suggestion
You may need to run it several times for it to fail again.
This error will not reproduce consistently because it is dependent on some variability in the execution times of the individual process calls.
```